tests/qinfo/test_trace_distance.py

```diff
@@ -0,0 +1,545 @@
+            ({"x": 0, "y": np.pi}), ({"x": 0, "y": 0})
+        )
+
+        assert qml.math.allclose(td_args, 0.0)
+        assert qml.math.allclose(td_arg_kwarg, 0.0)
+        assert qml.math.allclose(td_kwargs, 0.0)
+
+    # 0 and 2 * pi are removed to avoid nan values in the gradient
+    parameters = np.linspace(0, 2 * np.pi, 20)[1:-1]
+    wires = [1, 2]
+
+    @pytest.mark.autograd
+    @pytest.mark.parametrize("param", parameters)
+    @pytest.mark.parametrize("wire", wires)
+    def test_trace_distance_qnodes_rx_pauliz_grad(self, param, wire):
+        """Test the gradient of the trace distance between Rx and PauliZ circuits."""
+        dev = qml.device("default.qubit", wires=wire)
+
+        @qml.qnode(dev, interface="autograd", diff_method="backprop")
+        def circuit0(x):
+            qml.RX(x, wires=wire - 1)
+            return qml.state()
+
+        @qml.qnode(dev, interface="autograd", diff_method="backprop")
+        def circuit1():
+            qml.PauliZ(wires=wire - 1)
+            return qml.state()
+
+        td_grad = qml.grad(
+            qml.qinfo.trace_distance(circuit0, circuit1, wires0=[wire - 1], wires1=[wire - 1])
+        )((qml.numpy.array(param, requires_grad=True)))
+        expected_td = expected_grad_trace_distance_rx_pauliz(param)
+        assert qml.math.allclose(td_grad, expected_td)
+
+        td_grad = qml.grad(
+            qml.qinfo.trace_distance(circuit1, circuit0, wires0=[wire - 1], wires1=[wire - 1])
+        )(None, qml.numpy.array(param, requires_grad=True))
+        assert qml.math.allclose(td_grad, expected_td)
+
+    interfaces = ["auto", "autograd"]
+
+    @pytest.mark.autograd
+    @pytest.mark.parametrize("param", parameters)
+    @pytest.mark.parametrize("wire", wires)
+    @pytest.mark.parametrize("interface", interfaces)
+    def test_trace_distance_qnodes_rx_tworx_grad(self, param, wire, interface):
+        """Test the gradient of the trace distance between two trainable circuits."""
+        dev = qml.device("default.qubit", wires=wire)
+
+        @qml.qnode(dev, interface=interface, diff_method="backprop")
+        def circuit(x):
+            qml.RX(x, wires=wire - 1)
+            return qml.state()
+
+        td_grad = qml.grad(
+            qml.qinfo.trace_distance(circuit, circuit, wires0=[wire - 1], wires1=[wire - 1])
+        )(
+            (qml.numpy.array(param, requires_grad=True)),
+            (qml.numpy.array(2 * param, requires_grad=True)),
+        )
+        expected = expected_grad_trace_distance_rx_pauliz(param)
+        expected_td = [-expected, expected]
+        assert qml.math.allclose(td_grad, expected_td)
+
+    interfaces_func = [
+        ("auto", lambda x: x),
+        ("torch", torch.tensor),
+        ("tf", tf.Variable),
+        ("jax", jax.numpy.array),
+        ("jax-jit", jax.numpy.array),
+    ]
+
+    @pytest.mark.torch
+    @pytest.mark.tf
+    @pytest.mark.jax
```

Comment:
If you use `pytest.mark` with more than one interface marker, then it won't actually run on GitHub actions (even though it might run locally). You need instead mark this with `pytest.mark.all_interfaces`.